src/platform/OpenThread/GenericNetworkCommissioningThreadDriver.cpp

```diff
@@ -0,0 +1,191 @@
+/*
+ *
+ *    Copyright (c) 2021 Project CHIP Authors
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#include <lib/support/CodeUtils.h>
+#include <lib/support/SafeInt.h>
+#include <platform/CHIPDeviceLayer.h>
+#include <platform/NetworkCommissioning.h>
+#include <platform/OpenThread/GenericNetworkCommissioningThreadDriver.h>
+#include <platform/OpenThread/GenericThreadStackManagerImpl_OpenThread.h>
+#include <platform/ThreadStackManager.h>
+
+#include <limits>
+#include <string>
```

Comment:
linter complains about string and vector usage here and I believe it is correct. These files seem quite low level so they should probably not rely on heap too much.

Reply:
I will remove the string. Not sure of the best way to implement the `ScanResponseIterator` without vector. I will think of a way but open to suggestion.

Reply:
I ended up deleting OtScanResponseIterator therefore also the need for std::vector. It wasn't used yet. I will make that work in another PR to Handle ScanNetwork command and ScanResponse without the use of vector